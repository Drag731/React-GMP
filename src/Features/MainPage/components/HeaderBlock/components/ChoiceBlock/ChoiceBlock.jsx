// @flow
import React from 'react';

import Button from '../../../../../../components/Button/Button';

type Props = {
    handleSearchButton: Function,
    handleTitleSearch: Function,
    handleDirectorSearch: Function,
};

const ChoiceBlock = (props: Props) => {
    const {handleTitleSearch, handleDirectorSearch, handleSearchButton} = props;
    return (
        <div className="b-header__choice choice-flex">
            <div className="choice-flex__item">
                <span className="b-header__choice-text">search by</span>
                <Button handler={handleTitleSearch}>title</Button>
                <Button handler={handleDirectorSearch}>genres</Button>
            </div>
            <div className="choice-flex__item">
                <Button handler={handleSearchButton}>search</Button>
            </div>
        </div>
    )
};

export default ChoiceBlock;