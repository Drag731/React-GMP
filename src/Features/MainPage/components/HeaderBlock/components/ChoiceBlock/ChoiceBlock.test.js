import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ChoiceBlock from './ChoiceBlock.jsx'

configure({ adapter: new Adapter() });

describe('<ChoiceBlock  />', () => {
    it('Snapshot one', () => {
        const handleTitleSearch = jest.fn();
        const handleDirectorSearch = jest.fn();
        const tree = renderer
            .create(<ChoiceBlock
                handleTitleSearch={handleTitleSearch}
                handleDirectorSearch={handleDirectorSearch}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Renders ChoiceBlock component', () => {
        const handleTitleSearch = jest.fn();
        const handleDirectorSearch = jest.fn();
        const tree = shallow(
            <ChoiceBlock
                handleTitleSearch={handleTitleSearch}
                handleDirectorSearch={handleDirectorSearch}
            />
        );
        expect(tree.find('.b-header__choice')).toHaveLength(1);
    });
});
