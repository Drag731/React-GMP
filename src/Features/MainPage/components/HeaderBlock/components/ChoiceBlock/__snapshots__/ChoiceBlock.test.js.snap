// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChoiceBlock  /> Snapshot one 1`] = `
<div
  className="b-header__choice choice-flex"
>
  <div
    className="choice-flex__item"
  >
    <span
      className="b-header__choice-text"
    >
      search by
    </span>
    <button
      onClick={[MockFunction]}
    >
      title
    </button>
    <button
      onClick={[MockFunction]}
    >
      genres
    </button>
  </div>
  <div
    className="choice-flex__item"
  >
    <button
      onClick={[MockFunction]}
    >
      search
    </button>
  </div>
</div>
`;
